### ISSUE TITLE
Let game console "redial" ignore duplicated repeated commands

### ISSUE THREAD

**Issue by Cervator** [MovingBlocks/Terasology#2813]
**Current situation**: if you enter `give stone` once then `give dirt` 10 times in the game console (`F1` or `grave` / `~` key key) you need to `up arrow` 11 times before you'll get `give stone` from the console history again.

**Desired situation**: console history ignores successive identical commands (and also always ignores entirely empty commands)

**See**: `ConsoleImpl.java` for some of the involved code.

Should be easy!

**Comment by bhi5hmaraj** [comment]
Hi , I'm Bhishmaraj Computer Science Undergrad . I'm new to the open source community and I'm interested in fixing this bug . I have already cloned the project from GitHub and built it using gradlew and tried playing it . Can you please help get getting started to fix this bug.

**Comment by ThompsonTyler** [comment]
You will probably want to check out `UICommandEntry.java` for the "redialing" and then setup the command history to work as desired.

**Comment by bhi5hmaraj** [comment]
Thanks , I'll try to fix this

**Comment by emanuele3d** [comment]
@bhi5hmaraj: it appears vampcat has already tackled this successfully?

If so @vampcat, please be aware of other people who already offered to work on issues. We are a community and one of the things we try to do is to be considerate with each other.

**Comment by vampcat** [comment]
Ah, I'm sorry. Cervator offered up the task on IRC as soon as it was added, and I completed the task at that time. I thought I commited too, but apparently GitHub Desktop thought otherwise.. so the late commit. bhi4hmaraj had not started at that time. Either way, I'll make a comment next time before starting.

**Comment by Cervator** [comment]
Yeah I probably caused this one, as I realized moments after I advertised the new issue on IRC with several students active + it being the very first issue listed while other students are surfing our issue tracker ... :D

But yep if coding starts please note on the issue :-)

**Comment by vampcat** [comment]
Aye aye cap'n.

**Comment by vampcat** [comment]
Can someone mark this as closed?

**Comment by Cervator** [comment]
Whoops, missed this one. Thanks @vampcat!

Closing as complete
